src/decaf.cpp

```diff
@@ -210,6 +210,53 @@ namespace decaf {
     return ExecutionResult::Continue;
   }
 
+  ExecutionResult Interpreter::visitICmpInst(llvm::ICmpInst& icmp) {
+    using llvm::ICmpInst;
+
+    StackFrame& frame = ctx->stack_top();
+
+    auto lhs = normalize_to_int(frame.lookup(icmp.getOperand(0), *z3));
+    auto rhs = normalize_to_int(frame.lookup(icmp.getOperand(1), *z3));
+
+    switch (icmp.getPredicate()) {
```

Comment:
It looks like ICMP is supposed to implement vector comparison as well (by doing element wise comparison). It doesn't look like that's implemented?

Reply:
I'm pretty sure that was explicitly out of scope for the prototype? Nothing we have at this point supports vector operands